cli/src/commands/fix.rs

```diff
@@ -158,6 +159,11 @@ pub(crate) fn cmd_fix(
         .parse_file_patterns(ui, &args.paths)?
         .to_matcher();
 
+    // See #4866.
+    // Fix commands must run from the repo root, as it may read files such as
+    // .clang-format that depend on the working directory being correct.
+    let working_dir = workspace_command.repo_path().to_path_buf();
```

Comment:
Shouldn't it be `workspace_root()`? Maybe we can add a fake-formatter option that stat/read a file to test the behavior.

fwiw, this can be inlined as `tx.base_workspace_helper()...`.